server/jetstream_cluster.go

```diff
@@ -7893,7 +7893,8 @@ func (mset *stream) processClusteredInboundMsg(subject, reply string, hdr, msg [
 				pubAck := append(buf[:0], mset.pubAck...)
 				seq := dde.seq
 				mset.mu.Unlock()
-				if canRespond {
+				// Should not return an invalid sequence, in that case timeout.
+				if canRespond && seq > 0 {
```

Comment:
When will seq == 0 here?

I want to understand when a response now becomes a timeout, which can be a pretty big change in behavior.

Reply:
For the following scenario:
- two messages are published around the same time, both with the same `msgId`
- one message goes in, sets the `dde.seq=0` and gets proposed to the followers
- now the second message goes in, proposal hasn't gone through yet, that would timeout

If the proposal of the first message has gone through, the appropriate seq will be set for `dde.seq`, so then the second message would get a proper `seq=1, duplicate=true` for example.